### ISSUE TITLE
geomag-algorithms

### ISSUE THREAD

**Issue by NMoghaddam** [usgs/geomag-algorithms#282]
I need to install geomag-algorithms in order to have a python3 software package to run. The following command results in an error that I need help to resolve it:
pip install git+https://github.com/usgs/geomag-algorithms.git

  ERROR: Failed building wheel for obspy
  Running setup.py clean for obspy
Successfully built geomag-algorithms
Failed to build obspy
Installing collected packages: obspy, geomag-algorithms
    Running setup.py install for obspy ... error
    ERROR: Command errored out with exit status 1:
     command: /Users/neginmoghaddam/anaconda3/envs/pysar3.7/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/7f/pxpmf_bj1y57p35299g8mtyc0000gn/T/pip-install-h1s4oe45/obspy/setup.py'"'"'; __file__='"'"'/private/var/folders/7f/pxpmf_bj1y57p35299g8mtyc0000gn/T/pip-install-h1s4oe45/obspy/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/7f/pxpmf_bj1y57p35299g8mtyc0000gn/T/pip-record-6ta4yvdh/install-record.txt --single-version-externally-managed --compile --install-headers /Users/neginmoghaddam/anaconda3/envs/pysar3.7/include/python3.7m/obspy

I have a python3.7 environment that this geomag-algorithms need to be installed in it.

Thanks for the help.

**Comment by jmfee-usgs** [comment]
Hello, we recommend using the `conda-forge` channel to install obspy:
https://github.com/usgs/geomag-algorithms/blob/master/docs/install.md#installation

If that doesn't work, can you provide more information about your environment?  I know when installing obspy through other channels (where it is not pre-built) gcc is generally required...

**Comment by jmfee-usgs** [comment]
@NMoghaddam : I was able to reproduce this after installing into an empty environment.  It turns out this is related to a change in setuptools 46 that affects obspy ( https://github.com/obspy/obspy/pull/2578 ).

Until obspy 1.2.1 is released with a patch, you can install obspy by downgrading setuptools before installation:
```
pip install 'setuptools==45.3.0'
pip install git+https://github.com/usgs/geomag-algorithms.git
```

**Comment by megies** [comment]
I just released obspy 1.2.1 bug fix release, so this should work as intended again.

**Comment by NMoghaddam** [comment]
@jmfee-usgs Thanks alot for the response. I confirm with your hint on obsby installation, the problem is now resolved and I successfully installed and test the software package.
